integration-test/src/main/java/com/sequenceiq/it/cloudbreak/EnvironmentClient.java

```diff
@@ -53,7 +53,8 @@ public FlowPublicEndpoint flowPublicEndpoint() {
     @Override
     public <E extends Enum<E>, T extends WaitObject> T waitObject(CloudbreakTestDto entity, String name, Map<String, E> desiredStatuses,
             TestContext testContext, Set<E> ignoredFailedStatuses) {
-        return (T) new EnvironmentWaitObject(this, entity.getName(), entity.getCrn(), (EnvironmentStatus) desiredStatuses.get("status"));
+        return (T) new EnvironmentWaitObject(this, entity.getName(), entity.getCrn(), (EnvironmentStatus) desiredStatuses.get("status"),
+                (Set<EnvironmentStatus>) ignoredFailedStatuses);
```

Comment:
I don't really like this casting, it smells a bit. I did some experiment, and if you modify this:
```
public abstract class MicroserviceClient<C, I, E extends Enum<E>> extends Entity {
```
then you can have the abstract method like:
```
    public abstract <W extends WaitObject> W waitObject(CloudbreakTestDto entity, String name, Map<String, E> desiredStatuses,
            TestContext testContext, Set<E> ignoredFailedStatuses);
```
and the EnvironmentClient implementation would look like:
```
    @Override
    public <T extends WaitObject> T waitObject(CloudbreakTestDto entity, String name, Map<String, EnvironmentStatus> desiredStatuses,
            TestContext testContext, Set<EnvironmentStatus> ignoredFailedStatuses) {
        return (T) new EnvironmentWaitObject(this, entity.getName(), entity.getCrn(), desiredStatuses.get("status"));
    }

Reply:
But then the problem comes with FreeipaClient, where the status can be:
- UserSyncState
- OperationState
- Status
I guess then the freeipaClient should be thorn apart into 3.

Reply:
As far as I understand the work is still in progress on the PR. Is this right?

Reply:
@aszegedi yes, I am working on the suggestions of Misi.